examples/webrtc/main.go

```diff
@@ -111,13 +111,17 @@ func main() {
 		panic(err)
 	}
 
-	// Sets the LocalDescription, and starts our UDP listeners
+	// Set the local SessionDescription
 	err = peerConnection.SetLocalDescription(answer)
 	if err != nil {
 		panic(err)
 	}
 
+	//Wait for ICE gathering to complete (non-trickle ICE)
+	<-webrtc.GatheringCompletePromise(peerConnection)
```

Comment:
You need to create the promise before `SetLocalDescription` if gathering finishes really quickly the resolve can happen before you create the promise (then it will just block forever)

* Create Promise
* SetLocalDescription
* Wait for promise

Reply:
Cool! So the ICE gatherer will update the SDP at `*peerConnection.LocalDescription()` multiple times and it's good to block until it finds the best candidate?

Reply:
Hmmm still getting the same error

Reply:
Yea! Looking at the code that all looks right. That matches what we do in all our other examples.

I am AFK for a bit but will try tonight!

Reply:
Cool thank you!

Reply:
@Sean-Der wouldn't it be fine since we have this check, https://github.com/pion/webrtc/blob/v3.0.0/gathering_complete_promise.go#L15?